### PR TITLE
fix(nams): handle auto-merged entity creates in both SDKs; fix crewai smoke test

### PR DESCRIPTION
**Pull request by johnymontana**
Fixes the two failing tests in the NAMS integration CI run:

- `tests/integration/nams/test_tck_platinum.py::test_get_entity_history`
- `tests/integration/nams/test_framework_smoke.py::test_crewai_smoke`

## `add_entity` breaks when NAMS auto-merges the create (real client bug, both SDKs)

NAMS resolves-before-create: when a new entity name is sufficiently similar to an
existing entity, `POST /v1/entities` merges onto it and responds

```json
{"id": "...", "resolution": "merged", "merged_into": "...", "confidence": 0.93}
```

with no name/type (verified against the server's entity_create.go). The
platinum test tripped this because the shared sandbox accumulates similarly-named
itest-*-history-target-1 entities across CI runs — the failure was
sandbox-state-dependent, not specific to get_entity_history (the traceback's
add_entity call was the actual failure point).

- Python (src/neo4j_agent_memory/nams/long_term.py): parsing the merged
  response as an Entity raised pydantic.ValidationError. add_entity now
  detects resolution: "merged", follows up with GET /entities/{merged_into},
  and returns the canonical merged-into entity. If that read fails, it
  synthesizes a parseable Entity from the request's name/type plus the
  returned id and confidence.
- TypeScript (typescript/src/long-term/index.ts): same bug with a quieter
  failure mode — no runtime validation, so addEntity silently returned an
  Entity with undefined name and type. Same fix: follow-up getEntity,
  fallback on SDK errors (MemoryError subclasses), non-SDK errors still
  propagate.

created and review_pending responses carry full entity fields and are
unaffected (locked in by tests).

test_crewai_smoke constructed the adapter wrong (test bug)

The test called Neo4jCrewMemory(memory_client=nams_client) without the
required crew_id — the docs and tests/integration/test_crewai_integration.py
all pass it. The test now passes crew_id=f"{nams_session}-crew", and the stale
comment claiming the adapter uses an internal crew id is corrected.

Tests

- Python: three new respx-mocked unit tests in tests/unit/nams/test_long_term.py
  (merged → canonical fetch, merged + 404 → request-field fallback,
  review_pending → parses directly).
- TypeScript: new typescript/test/unit/add-entity-merged.test.ts mirroring the
  above plus a non-SDK-error rethrow case.
- Changelog entries added under Unreleased in both CHANGELOG.md files.

Verification

- Python: full unit suite passes (1418 passed); mypy --strict, ty, and ruff
  clean on the changed files; both integration files collect cleanly (live runs
  need sandbox credentials).
- TypeScript: tsc --noEmit clean, npm test passes (162 tests / 25 files),
  npm run build succeeds — the same jobs the TS PR CI runs. The bridge TCK
  suite is env-gated and unaffected (bridge responses never carry
  resolution: "merged").

### CODE REVIEW

File: src/neo4j_agent_memory/nams/long_term.py [hunk 1]
```diff
@@ -227,8 +240,44 @@ async def add_entity(
...
+        from the request so the caller still gets a parseable Entity.
+        """
+        merged_id = payload.get("merged_into") or payload.get("mergedInto") or payload.get("id")
+        if merged_id:
+            try:
+                detail = await self._transport.request(
+                    _SPEC_GET_ENTITY,
+                    path_params={"entity_id": _to_str(merged_id)},
+                )
+            except MemoryError:
```
Comment (Andy2003) [inline src/neo4j_agent_memory/nams/long_term.py:268]: `MemoryError` is the root of the NAMS error tree, so this also swallows `AuthenticationError` (401/403), `RateLimitError` (429) and `TransportError` (5xx/network). Only 404 maps to a bare `MemoryError` (`transport.py:358`).

Effect: a rate-limited or auth-failed canonical read returns the synthesized entity with the *requested* name — no error, no log.

Needs a `NotFoundError` in `core/exceptions.py`, or `except MemoryError as exc:` + `if type(exc) is not MemoryError: raise`.

File: src/neo4j_agent_memory/nams/long_term.py [hunk 2]
```diff
@@ -227,8 +240,44 @@ async def add_entity(
...
+                detail = None
+            if isinstance(detail, dict) and detail.get("name"):
+                return detail
+        return _drop_none(
+            {
+                "id": merged_id,
+                "name": name,
+                "type": nams_type,
+                "confidence": payload.get("confidence"),
```
Comment (Andy2003) [inline src/neo4j_agent_memory/nams/long_term.py:277]: Reusing the merge score as `Entity.confidence`:

1. It is a name-similarity score, not extraction confidence. The canonical-fetch path returns the entity's own value, so the field means different things per path — a caller filtering `confidence >= 0.95` drops a canonical entity whose stored value is 1.0.
2. `Entity.confidence` is `Field(ge=0.0, le=1.0)`; an out-of-range score makes this branch raise the `ValidationError` it exists to prevent.

Drop it (model default 1.0) or move to `metadata`.

File: src/neo4j_agent_memory/nams/long_term.py [hunk 3]
```diff
@@ -167,10 +167,16 @@ def _normalize_entity(payload: dict[str, Any] | None) -> dict[str, Any]:
...
+
+    Explicit ``null`` fields are dropped so model defaults apply — NAMS
+    projects unset node properties as JSON ``null`` (e.g. ``confidence``
+    on a manually-created entity), which would otherwise fail parsing
+    for non-optional fields like ``confidence: float``.
     """
     from datetime import datetime, timezone
 
     data = snakeize_keys(payload) if isinstance(payload, dict) else {}
+    data = {k: v for k, v in data.items() if v is not None}
```
Comment (Andy2003) [inline src/neo4j_agent_memory/nams/long_term.py:179]: Runs on every entity parse, not just merges, and strips identity fields.

- `"id": null` -> client-side UUID via `MemoryEntry.id`'s `default_factory=uuid4`. `search_entities` can then return entities whose id does not exist server-side; a later `get_entity_relationships(entity.id)` 404s with no clue why.
- `"created_at": null` -> silently `now()`.

Restrict to fields with meaningful defaults (`confidence`, `description`, `subtype`, `updated_at`).

File: src/neo4j_agent_memory/nams/long_term.py [hunk 4]
```diff
@@ -227,8 +240,44 @@ async def add_entity(
             }
         )
         payload = await self._transport.request(_SPEC_ADD_ENTITY, json=body)
+        if isinstance(payload, dict) and payload.get("resolution") == "merged":
+            payload = await self._resolve_merged_entity(payload, name=name, nams_type=body["type"])
```
Comment (Andy2003) [inline src/neo4j_agent_memory/nams/long_term.py:244]: `resolution` / `merged_into` / merge confidence are discarded — caller asks for "Alice Smith", gets "Alice" under a different id, no signal.

`_normalize_entity` already defaults `metadata` to `{}`; `{"resolution": "merged", "merged_into": ..., "merge_confidence": ...}` there is non-breaking.

File: typescript/src/long-term/index.ts [hunk 5]
```diff
@@ -192,13 +207,48 @@ export class LongTermMemory {
...
+    wire: WireMergedResolution,
+    name: string,
+    entityType: string,
+  ): Promise<Entity> {
+    const mergedId = wire.merged_into ?? wire.id;
+    if (mergedId) {
+      try {
+        return await this.getEntity(mergedId);
+      } catch (error) {
+        if (!(error instanceof MemoryError)) throw error;
```
Comment (Andy2003) [inline typescript/src/long-term/index.ts:242]: Same over-broad catch as Python: `MemoryError` is the base class, so `AuthenticationError` / `ValidationError` / `TransportError` all fall through to the synthesized entity.

Caveat on the obvious fix: `NotFoundError` (`errors.ts:44`) is exported but never thrown in `src/` — `rest.ts:622-648` throws `TransportError` with `status` for every non-ok response, 404 included. Narrowing to `NotFoundError` disables the fallback entirely. Use `error instanceof TransportError && error.status === 404`.

File: typescript/src/long-term/index.ts [hunk 6]
```diff
@@ -192,13 +207,48 @@ export class LongTermMemory {
...
+   */
+  private async resolveMergedEntity(
+    wire: WireMergedResolution,
+    name: string,
+    entityType: string,
+  ): Promise<Entity> {
+    const mergedId = wire.merged_into ?? wire.id;
+    if (mergedId) {
+      try {
+        return await this.getEntity(mergedId);
```
Comment (Andy2003) [inline typescript/src/long-term/index.ts:240]: Missing Python's guard (`long_term.py:270`, `if isinstance(detail, dict) and detail.get("name")`): a 200 whose body lacks `name` returns `{name: undefined}` — the bug this PR fixes, one hop later.

Sharper: `rest.ts:615` (204) and `rest.ts:652` (empty body) return `undefined`, so `getEntity` -> `toEntity(undefined)` throws a raw `TypeError` reading `.id`. Not a `MemoryError`, so it escapes the catch below and propagates out of `addEntity`.

File: typescript/src/long-term/index.ts [hunk 7]
```diff
@@ -192,13 +207,48 @@ export class LongTermMemory {
...
+      } catch (error) {
+        if (!(error instanceof MemoryError)) throw error;
+        // Canonical record unreadable — fall through to the synthesized form.
+      }
+    }
+    return toEntity({
+      id: mergedId ?? "",
+      name,
+      type: entityType,
```
Comment (Andy2003) [inline typescript/src/long-term/index.ts:249]: `entityType` verbatim diverges from every other path, which returns the server's lowercase echo: `addEntity("x", "PERSON")` gives `type: "person"` on create and on merged+GET, but `"PERSON"` here. Python passes the already-mapped `body["type"]` through the same normalization.

Also `createdAt` lands as `""` (`toEntity`, line 135) -> `new Date(entity.createdAt)` is `Invalid Date`. Python synthesizes `datetime.now(timezone.utc)`.

File: typescript/src/long-term/index.ts [hunk 8]
```diff
@@ -108,20 +120,23 @@ interface WireGraph {
...
+    canonicalName: w.canonical_name ?? undefined,
     createdAt: w.created_at ?? "",
-    updatedAt: w.updated_at,
-    confidence: w.confidence,
-    sourceStage: w.source_stage,
-    relationships: w.relationships?.map(toRelRef),
+    updatedAt: w.updated_at ?? undefined,
+    confidence: w.confidence ?? undefined,
+    sourceStage: w.source_stage ?? undefined,
+    relationships: w.relationships?.map(toRelRef) ?? undefined,
```
Comment (Andy2003) [inline typescript/src/long-term/index.ts:139]: Normalization stops at `toEntity` — `toRelRef` (and `toPreference` / `toFact` / `toMention`) still pass nulls through. The merged-entity GET returns inlined `relationships`, so nulls leak on exactly the path this PR adds: `EntityRelationshipRef.targetName` / `properties` can be `null` despite being typed `| undefined`.

**Review by Andy2003 — COMMENTED** [review]
Merged-create handling works; findings below are uncovered edge cases. Verified at head: `tests/unit/nams` 380 passed, `tsc --noEmit` clean, `vitest run test/unit` 108 passed. The `crew_id` fix is correct (`crew_id` is a required param).
